runtime/shared_heap.c

```diff
@@ -47,8 +47,10 @@ typedef struct pool {
   value* next_obj;
   caml_domain_state* owner;
   sizeclass sz;
+  uintnat padding[28];
```

Comment:
For the record: what this padding is doing (and the `#define POOL_HEADER_WSIZE 32`) is to ensure that allocations in the pool start at a 256-aligned address on 64-bit platforms and 128-aligned address on 32-bit platforms.   s390x being a 64-bit platform, this alignment is sufficient, although more than strictly necessary on other platforms.